server.mjs

```diff
@@ -418,19 +502,26 @@ MinifyHTML().then(function(result) {
     }).catch(next);
   });
 
-  router.put('/asset/:link', async function(req, res) {
+  router.put('/asset/:link', apiLimiter, async function(req, res) {
     try {
+      if (assetCache[req.params.link]) {
+        res.send(`/assets${assetCache[req.params.link]}`);
+        return;
+      }
       const content = Buffer.from(await (await fetch(req.params.link)).arrayBuffer());
       const filename = `/${CRC32.buf(content)}_${content.length}`;
-      if(!Config.resolveAsset(filename.substr(1)))
+      if(!Config.resolveAsset(filename.substr(1))) {
         fs.writeFileSync(assetsdir + filename, content);
+      }
+      assetCache[req.params.link] = filename;
+      fs.writeFileSync(assetsdir + '/cache.json', JSON.stringify(assetCache));
```

Comment:
Was this a problem in this PR? This saves a cache of already downloaded assets that were given by link, right?

Reply:
This was added by the changes made adding the apilimiter.